### PR TITLE
[RDY] Add basic note support to constructions.

### LINKED ISSUE #1385: Palisade Gate doesn't play well with others
Apparently the Palisade Gate requires Palisade Walls on either side in order to work.  This wasn't obvious to at least one player, and it's a pretty expensive lesson to learn through construction fail.  Seems worth either making a note in the menu or loosening the restriction.

See http://smf.cataclysmdda.com/index.php?topic=14.msg19238#msg19238 .
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/1366853-palisade-gate-doesn-t-play-well-with-others?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by kevingranade**
Fixes #1385 (LOOKIT THAT NUMBER!!!! HOW DID WE NOT DO THIS ALREADY?!?!?)
 
I'm open to suggestions for improving the note, I couldn't think of better wording, but maybe you can.  Also can take suggestions for constructions that need notes, please include the notes, not just "foo is unclear".

Translation support added.
Also includes the fix to extract morale_type strings from #20527

### CODE REVIEW

File: data/json/construction.json [hunk 1]
```diff
@@ -1245,6 +1246,7 @@
             [ [ "2x4", 3 ] ],
             [ [ "rope_6", 2 ], [ "vine_30", 1 ] ]
         ],
+        "pre_note" : "Must be between palisade walls to function, at least one wall must have an adjacent rope & pulley system.",
```
Comment (Leland) [inline data/json/construction.json]: Note itself looks good – grammatically, you should either change the period to a comma, or add an "add" after it.

**Comment by kolsurma** [comment]
Probably because so few people use construction, vehicles are just that much better.

File: src/construction.cpp [hunk 2]
```diff
@@ -458,6 +458,12 @@ void construction_menu()
                             current_buffer.insert( current_buffer.end(), folded_result_string.begin(),
                                                    folded_result_string.end() );
                         }
+                        if( !current_con->pre_note.empty() ) {
+                            std::vector<std::string> folded_result_string =
+                                foldstring( current_con->pre_note, available_window_width );
```
Comment (BrettDong) [inline src/construction.cpp]: Is this localized?

Comment (kevingranade) [inline src/construction.cpp]: That was 'TODO', done now.

**Pull request merged**